threescale/src/proxy.rs

```diff
@@ -92,15 +96,79 @@ pub fn set_app_id_to_cache(user_key: &UserKey, app_id: &AppId) -> Result<(), Cac
 // returns false on set failure
 pub fn set_application_to_cache(key: &str, app: &Application, cas: u32) -> bool {
     info!("setting application with key: {}", key);
-    match set_shared_data(
-        key,
-        Some(&bincode::serialize::<Application>(app).unwrap()),
-        Some(cas),
-    ) {
-        Ok(()) => true,
-        Err(e) => {
-            debug!("set operation failed for key: {} : {:?}", key, e);
-            false
+    let prev_memory_usage = get_cache_pair_size(key) as i32;
+    let serialized_app = bincode::serialize::<Application>(app).unwrap();
```

Comment:
What would we need to do if this failed to serialize? Log something and return `false`?

Reply:
Handled